projects/scripts/adi_board.tcl

```diff
@@ -1089,19 +1125,25 @@ proc ad_cpu_interconnect {p_address p_name {p_intf_name {}}} {
     }
   }
 
-  set_property CONFIG.NUM_MI $sys_cpu_interconnect_index [get_bd_cells axi_cpu_interconnect]
+  if {($p_sel eq "HPM0_FPD")} {
+    set_property CONFIG.NUM_MI $sys_hpm0fpd_interconnect_index [get_bd_cells $interconnect_name]
+  } elseif {($p_sel eq "HPM1_FPD")} {
+    set_property CONFIG.NUM_MI $sys_hpm1fpd_interconnect_index [get_bd_cells $interconnect_name]
+  } elseif {($p_sel eq "HPM0_LPD")} {
+    set_property CONFIG.NUM_MI $sys_cpu_interconnect_index [get_bd_cells $interconnect_name]
+  }
 
   if {$use_smart_connect == 0} {
-    ad_connect sys_cpu_clk axi_cpu_interconnect/${i_str}_ACLK
-    ad_connect sys_cpu_resetn axi_cpu_interconnect/${i_str}_ARESETN
+    ad_connect sys_cpu_clk ${interconnect_name}/${i_str}_ACLK
```

Comment:
Don't use curly braces where it is not needed
(comment valid for all lines that interconnect_name is inside curly braces).